src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts

```diff
@@ -293,8 +294,9 @@ export class InlineChatWidget {
 			const widgetToolbarWidth = getTotalWidth(this._elements.widgetToolbar);
 			const innerEditorWidth = widgetDim.width - widgetToolbarWidth;
 			const inputDim = new Dimension(innerEditorWidth, this._inputWidget.getPreferredSize().height);
-			if (!this._lastDim || !Dimension.equals(this._lastDim, widgetDim)) {
+			if (!this._lastDim || !Dimension.equals(this._lastDim, widgetDim) || !this._lastInputDim || !Dimension.equals(this._lastInputDim, inputDim)) {
```

Comment:
When the toolbar updates later than the widget, we didn't get a chance to trigger layout again it seems. Thus I added the cache for the input to ensure that it can get updated correctly. cc @jrieken 

![image](https://github.com/microsoft/vscode/assets/876920/27ecac90-d38a-43a6-9b17-2e208797c9c7)